internal/jobqueue/fsjobqueue/fsjobqueue.go

```diff
@@ -581,3 +600,108 @@ func jobMatchesCriteria(j *job, acceptedJobTypes []string, acceptedChannels []st
 
 	return contains(acceptedJobTypes, j.Type) && contains(acceptedChannels, j.Channel)
 }
+
+func (q *fsJobQueue) getJsonField(data *json.RawMessage, field string, result *json.RawMessage) error {
+	m := make(map[string]json.RawMessage)
+	if len(*data) > 0 {
+		err := json.Unmarshal(*data, &m)
+		if err != nil {
+			return err
+		}
+		*result = m[field]
+	}
+	return nil
+}
+
+func (q *fsJobQueue) QueryResultFields(id uuid.UUID, paths []string, response any) error {
+	for _, path := range paths {
+		err := jobqueue.FieldSanitazation(path)
+		if err != nil {
+			return err
+		}
+		hasField, err := q.TestResultFieldExists(id, path)
+		if err != nil {
+			return err
+		}
+		if hasField {
+			// if recipient is a pointer to something, the something in question will get allocated by the
+			// FindRecipientByTagPath function. So only call this when there is an actual result to unpack, otherwise
+			// the content will be set to non nil (and that is a problem).
+			recipient, err := jobqueue.FindRecipientByTagPath(response, path)
+			if err != nil {
+				return err
+			}
+			err = q.queryResultField(id, path, recipient)
```

Comment:
If I get the code right, this method reads the whole job JSON file and (partially) unmashals it. We are in a loop, though, thus this whole process will be done multiple times.

I honestly wonder if we shouldn't just read the JSON once, unmarshal it directly into `results` while ignoring `paths`. The whole idea of `fsjobqueue` is that it isn't optimized for speed, but for simplicity, because it's not meant to be used in high-load environments. It's for the cases when you don't want to set up an external database for composer (i.e. when you install composer locally on your machine).